query/query.go

```diff
@@ -2383,6 +2398,35 @@ func ProcessGraph(ctx context.Context, sg, parent *SubGraph, rch chan error) {
 	rch <- childErr
 }
 
+// applies "every" to lists inside uidMatrix
+// every 'every-th' uid is selected from the uid lists,
+// padding uids are considered when skipping uids from next list
+func (sg *SubGraph) applyEvery(ctx context.Context) error {
+	sg.updateUidMatrix()
+
+	offset := 0
+	every := sg.Params.Every
+	for i := 0; i < len(sg.uidMatrix); i++ {
+		uidList := sg.uidMatrix[i].Uids
```

Comment:
@mangalaman93 assumption here is that uids in `sg.uidMatrix[i].Uids` are sorted ascending order, and all uids in `sg.uidMatrix[i].Uids` are smaller than in `sg.uidMatrix[i+1].Uids`. Is that true?

Reply:
No, i+1 doesn't have to be bigger than i. We merge these later using MergeSorted. Uids should be in the correct order.

Reply:
So the individual `sg.uidMatrix[*].Uids` lists are arbitrary subsets of the entire uid space. This renders the "offset" logic obsolete.

Any idea how to efficiently sample every `every`-th uid deterministically?

Reply:
Given https://github.com/dgraph-io/dgraph/pull/8977/files#diff-c6a819ee05eb6643352fb889fdd48569cca191565c59c96ec849150a12bad2feR2417, it looks like then notion of having multiple `sg.uidMatrix[*].Uids lists is different to what I presume. What does it mean to have multiple `uidMatrix`?

Reply:
I have looked through the documentation of `subGraph` and I think I understand `sd.uidMatrix` now. Updating tests.